invader_payment_stripe/services/payment_stripe.py

```diff
@@ -120,32 +124,36 @@ def confirm_payment(self, target, **params):
             * The stripe_payment_intent_id is passed
             * The intent state is 'succeeded'
         :param target: string (authorized value is checked by service)
-        :param payment_mode: string (The Odoo payment mode id)
+        :param payment_mode_id: string (The Odoo payment mode id)
         :param stripe_payment_method_id:
         :param stripe_payment_intent_id:
         :return:
         """
-        payment_mode = params.get("payment_mode")
+        payment_mode_id = params.get("payment_mode_id")
         stripe_payment_method_id = params.get("stripe_payment_method_id")
         stripe_payment_intent_id = params.get("stripe_payment_intent_id")
         transaction_obj = self.env["payment.transaction"]
         payable = self.component(
             usage="invader.payment"
         )._invader_find_payable_from_target(target, **params)
+
         # Stripe part
         transaction = None
+        payment_mode = self.env["account.payment.mode"].browse(payment_mode_id)
+
+        acquirer = payment_mode.payment_acquirer_id.sudo()
+        if acquirer.provider != "stripe":
+            raise UserError(_("Payment mode acquirer mismatch."))
```

Comment:
@sebastienbeau The error message doesn't explain the problem... The issue here is that the stripe service is called with a payment_mode  linked to a non stripe acquirer... (xyz != stripe)

Reply:
Payment mode acquirer mismatch => Payment mode acquirer mismatch, should be 'stripe' found 'transfer' ?

Reply:
Payment mode acquirer mismatch, should be 'stripe' instead of 'transfer'